snmp/datadog_checks/snmp/snmp.py

```diff
@@ -98,13 +106,49 @@ def _get_instance_key(self, instance):
 
         return key
 
+    def discover_instances(self):
+        network = ipaddress.ip_network(self._config.network_address)
+        discovery_interval = self._config.instance.get('discovery_interval', 3600)
+        discovery_pause = self._config.instance.get('discovery_pause', 1)
+        while self._running:
+            start_time = time.time()
+            for host in network.hosts():
+                host = str(host)
+                if host in self._config.discovered_instances:
+                    continue
+                instance = self._config.instance.copy()
+                instance.pop('network_address')
+                instance['ip_address'] = host
+
+                config = InstanceConfig(
+                    instance,
+                    self.warning,
+                    self.init_config.get('global_metrics', []),
+                    self.mibs_path,
+                    self.profiles,
+                    self.profiles_by_oid,
+                )
+                try:
+                    sys_object_oid = self.fetch_sysobject_oid(config)
+                except Exception:
+                    continue
+                if sys_object_oid not in self.profiles_by_oid:
+                    self.log.warn("Host %s didn't match a profile for sysObjectID %s", host, sys_object_oid)
+                profile = self.profiles_by_oid[sys_object_oid]
+                config.refresh_with_profile(self.profiles[profile], self.warning)
+                self._config.discovered_instances[host] = config
+                time.sleep(discovery_pause)
```

Comment:
Why is `time.sleep(discovery_pause)` necessary ?

Reply:
It may be a bit aggressive to go through all hosts without pause. It get be set to 0 if useless.

Reply:
I don't really understand why it's an issue to go through all the hosts without pause on startup (at least I don't understand why so long, 1 sec is very long).

Not sure why/if we should provide `discovery_pause` config.

Reply:
Removed it.